src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ImplementGenericMathInterfacesCorrectlyTests.cs

```diff
@@ -0,0 +1,658 @@
+{ }
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMIRuleForInterface).WithLocation(0).WithArguments("IShiftOperators<TSelf, TOther, TResult>", "TSelf", "IMyInterface")).RunAsync();
+        }
+
+        [Fact]
+        public async Task IUnaryNegationOperatorsNotImplementedCorrectly()
+        {
+            await PopulateTestCs(@"
+using System.Numerics;
+
+interface IMyInterface : IUnaryNegationOperators<{|#0:int|}, int> // The 'IUnaryNegationOperators<TSelf, TResult>' interface requires the 'TSelf' type parameter to be filled with the derived type 'IMyInterface' 
+{ }
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMIRuleForInterface).WithLocation(0).WithArguments("IUnaryNegationOperators<TSelf, TResult>", "TSelf", "IMyInterface")).RunAsync();
+        }
+
+        [Fact]
+        public async Task IUnaryPlusOperatorsNotImplementedCorrectly()
+        {
+            await PopulateTestCs(@"
+using System.Numerics;
+
+interface IMyInterface : IUnaryPlusOperators<{|#0:int|}, int> // The 'IUnaryPlusOperators<TSelf, TResult>' interface requires the 'TSelf' type parameter to be filled with the derived type 'IMyInterface' 
+{ }
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMIRuleForInterface).WithLocation(0).WithArguments("IUnaryPlusOperators<TSelf, TResult>", "TSelf", "IMyInterface")).RunAsync();
+        }
+
+        [Fact]
+        public async Task ISubtractionOperatorsNotImplementedCorrectly()
+        {
+            await PopulateTestCs(@"
+using System.Numerics;
+
+interface IMyInterface : ISubtractionOperators<{|#0:int|}, int, int> // The 'ISubtractionOperators<TSelf, TOther, TResult>' interface requires the 'TSelf' type parameter to be filled with the derived type 'IMyInterface' 
+{ }
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMIRuleForInterface).WithLocation(0).WithArguments("ISubtractionOperators<TSelf, TOther, TResult>", "TSelf", "IMyInterface")).RunAsync();
+        }
+
+        [Fact]
+        public async Task IUnsignedNumberNotImplementedCorrectly()
+        {
+            await PopulateTestCs(@"
+using System.Numerics;
+
+interface IMyNumber : IUnsignedNumber<{|#0:uint|}> // The 'IUnsignedNumber<TSelf>' interface requires the 'TSelf' type parameter to be filled with the derived type 'IMyNumber' 
+{ }
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMIRuleForInterface).WithLocation(0).WithArguments("IUnsignedNumber<TSelf>", "TSelf", "IMyNumber")).RunAsync();
+        }
+
+        [Fact]
+        public async Task IFloatingPointConstantsNotImplementedCorrectly()
+        {
+            await PopulateTestCs(@"
+using System.Numerics;
+
+interface IMyNumber : IFloatingPointConstants<{|#0:double|}> // The 'IFloatingPointConstants<TSelf>' interface requires the 'TSelf' type parameter to be filled with the derived type 'IMyNumber' 
```

Comment:
`IFloatingPointConstants<TSelf>` is not included in the list, though the analyzer still flags it as it is derived from an interface. Similar interfaces added in the future no need to be added to the list in case they are derived from the interfaces in the list